# no-and-self

Disallows the [`.andSelf`](https://api.jquery.com/andSelf/) method. Prefer `.addBack`.


## Rule details

‚ùå Examples of **incorrect** code:
```js
$( 'div' ).andSelf( '.foo' );
$div.andSelf( '.foo' );
$( 'div' ).first().andSelf();
$( 'div' ).append( $( 'input' ).andSelf() );
```

‚úîÔ∏è Examples of **correct** code:
```js
andSelf();
[].andSelf();
div.andSelf();
div.andSelf;
```

üîß Examples of code **fixed** by this rule:
```js
$( 'div' ).andSelf( '.foo' );                /* ‚Üí */ $( 'div' ).addBack( '.foo' );
$div.andSelf( '.foo' );                      /* ‚Üí */ $div.addBack( '.foo' );
$( 'div' ).first().andSelf();                /* ‚Üí */ $( 'div' ).first().addBack();
$( 'div' ).append( $( 'input' ).andSelf() ); /* ‚Üí */ $( 'div' ).append( $( 'input' ).addBack() );
```

## Resources

* [Rule source](/src/rules/no-and-self.js)
* [Test source](/tests/rules/no-and-self.js)